eradicate.py

```diff
@@ -28,176 +28,199 @@
 import io
 import os
 import re
-import sys
 import tokenize
 
 __version__ = '1.0'
 
+class Eradicator:
+    """Eradicate comments."""
+    MULTILINE_ASSIGNMENT_REGEX = re.compile(r'^\s*\w+\s*=.*[(\[{]$')
+    PARTIAL_DICTIONARY_REGEX = re.compile(r'^\s*[\'"]\w+[\'"]\s*:.+[,{]\s*$')
+
+    DEFAULT_WHITELIST = [
+        r'pylint',
+        r'pyright',
+        r'noqa',
+        r'type:\s*ignore',
+        r'fmt:\s*(on|off)',
+        r'TODO',
+        r'FIXME',
+        r'XXX'
+    ]
+    WHITELIST_REGEX = re.compile(r'|'.join(DEFAULT_WHITELIST), flags=re.IGNORECASE)
+
+    def comment_contains_code(self, line, aggressive=True):
+        """Return True comment contains code."""
+        line = line.lstrip()
+        if not line.startswith('#'):
+            return False
+
+        line = line.lstrip(' \t\v\n#').strip()
+
+        # Ignore non-comment related hashes. For example, "# Issue #999".
+        if re.search('#[0-9]', line):
+            return False
+
+        # Ignore whitelisted comments
+        if self.WHITELIST_REGEX.search(line):
+            return False
+
+        if re.match(r'.*?coding[:=][ \t]*([-_.a-zA-Z0-9]+)', line):
+            return False
+
+        # Check that this is possibly code.
+        for symbol in list('()[]{}:=%') + ['print', 'return', 'break', 'continue',
+                                           'import']:
+            if symbol in line:
+                break
+        else:
+            return False
 
-MULTILINE_ASSIGNMENT_REGEX = re.compile(r'^\s*\w+\s*=.*[(\[{]$')
-PARTIAL_DICTIONARY_REGEX = re.compile(r'^\s*[\'"]\w+[\'"]\s*:.+[,{]\s*$')
-
-
-def comment_contains_code(line, aggressive=True):
-    """Return True comment contains code."""
-    line = line.lstrip()
-    if not line.startswith('#'):
-        return False
-
-    line = line.lstrip(' \t\v\n#').strip()
-
-    # Ignore non-comment related hashes. For example, "# Issue #999".
-    if re.search('#[0-9]', line):
-        return False
-
-    if line.startswith('pylint:'):
-        return False
+        if self.multiline_case(line, aggressive=aggressive):
+            return True
 
-    if re.match(r'.*?coding[:=][ \t]*([-_.a-zA-Z0-9]+)', line):
-        return False
+        symbol_list = [r'elif\s+.*', 'else', 'try',
```

Comment:
Let's move to a class level too

Reply:
I moved the `symbol_list` into constant `CODE_KEYWORDS` and also added constant `CODE_KEYWORDS_AGGR` for the aggressive version https://github.com/Cielquan/pydusting/blob/8b4a78091132e4016aeafc56b4c4d8cd13444cd5/eradicate.py#L84

I made it into another constant because its a class constant and if you would like to run the same instance multiple times for whatever reason and the first run is aggressive the second would be too because the value gets appended. And I also thinks it's for a better overview.